dashboard/lti_new.py

```diff
@@ -228,12 +246,24 @@ def extract_launch_variables_for_tool_use(request, message_launch):
         # update
         user_obj.first_name = first_name
         user_obj.last_name = last_name
+        user_obj.name = full_name
         user_obj.email = email
         user_obj.save()
     except User.DoesNotExist:
         password = ''.join(random.sample(string.ascii_letters, settings.RANDOM_PASSWORD_DEFAULT_LENGTH))
         user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name,
                                             last_name=last_name)
+
+    # Add username into the MyLA User table, since the data was not pulled in from cron job
+    user_id = CANVAS_DATA_ID_INCREMENT + int(canvas_user_id)
+    try:
+        myla_user_obj = MyLAUser.objects.get(user_id=user_id)
+        # update
+        myla_user_obj.sis_name = username
+        myla_user_obj.save()
+    except MyLAUser.DoesNotExist:
+        logger.warn(f'Cannot find MyLA user {username}')
```

Comment:
Is there a functional need for `sis_name`?

Reply:
yes, it is used in db_util.get_user_courses_info(), line 119, to determine which courses the user is member of

Reply:
I guess it's used a bunch of places. In an ideal world, we'd refactor this a bit so `auth_user` stores this data and `User` becomes `Enrollment`. That's already been discussed in the past. I guess this may work in the interim.

Reply:
True. The User table name is not ideal. We can do the table renaming in the future releases. We may not need the User.name field either. So that can be removed. 

I agree this is an interim solution to get the UDP query switch happen.